src/features/unified-share-modal/utils/convertData.js

```diff
@@ -263,37 +267,80 @@ export const convertCollabsResponse = (
     if (!entries.length) return { collaborators: [] };
 
     const ownerEmailDomain = ownerEmail && /@/.test(ownerEmail) ? ownerEmail.split('@')[1] : null;
-    const collaborators = entries.map(collab => {
-        const {
-            accessible_by: { id: userID, login: email, name, type },
-            id: collabID,
-            expires_at: executeAt,
-            role,
-        } = collab;
-        const collabEmailDomain = email.split('@')[1];
-        // Only display external collaborator icons if the current user owns the item
-        // and if the collaborator's email domain differs from the owner's email domain
-        const isExternalCollab = isCurrentUserOwner && collabEmailDomain !== ownerEmailDomain;
-        const convertedCollab: collaboratorType = {
-            collabID: parseInt(collabID, 10),
-            email,
-            hasCustomAvatar: false, // to do: connect to Avatar API
-            imageURL: null, // to do: connect to Avatar API
-            isExternalCollab,
-            name,
-            translatedRole: `${role[0].toUpperCase()}${role.slice(1)}`, // capitalize the user's role
-            type,
-            userID: parseInt(userID, 10),
-        };
-        if (executeAt) {
-            convertedCollab.expiration = { executeAt };
+
+    const collaborators = [];
+    entries.forEach(collab => {
+        // Only show accepted collaborations
+        if (collab.status && collab.status === STATUS_ACCEPTED) {
+            const {
+                accessible_by: { id: userID, login: email, name, type },
+                id: collabID,
+                expires_at: executeAt,
+                role,
+            } = collab;
+            const collabEmailDomain = email ? email.split('@')[1] : null;
```

Comment:
i see some email parsing logic in this and nearby lines. should this be collected into 1-2 utility functions?

Reply:
Some email validating/parsing functions here: https://github.com/box/box-ui-elements/blob/master/src/utils/parseEmails.js & https://github.com/box/box-ui-elements/blob/master/src/utils/validators.js